### PR TITLE
added some testing for EEAAQeRepv2 testing routine

### PR DESCRIPTION
**Pull request by jgriesfeller**
This is a try to run some simple testing on the EEA dataset.
I did not create anything in [conftest.py](https://github.com/metno/pyaerocom/blob/main-dev/tests/conftest.py) since I wanted to keep things simple.
I just wrote over the constants from the reading class that finds the data files to limit the file search when running on PPI. In the test data there's only two files right now anyway (besides the metadata file).
The test is limited to one variable so far. 

Shall I add all variables?

### CODE REVIEW

**Comment by jgriesfeller** [comment]
(since I noticed that the CI testing does not work)
what is a bit strange is that the local testing just works every other time:
```(pyadev) jang@pc5378:~/.../Python3/pyaerocom$ pytest tests/io/test_read_eea_aqerep_v2.py
==================================================================== test session starts =====================================================================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: /home/jang/data/Python3/pyaerocom, configfile: pytest.ini
plugins: anyio-3.3.0
collected 2 items                                                                                                                                            
run-last-failure: 1 known failures not in selected tests

tests/io/test_read_eea_aqerep_v2.py .F                                                                                                                 [100%]

========================================================================== FAILURES ==========================================================================
_________________________________________________________________________ test_read __________________________________________________________________________

reader = ReadEEAAQEREP_V2

    @testdata_unavail
    def test_read(reader):
        from pyaerocom.ungriddeddata import UngriddedData
        from pyaerocom.stationdata import StationData
    
        # special station codes to test
        # not sure if these are really needed
    
        # station ids to test
        station_id = {}
        # respective mean for a station; index has to be the same as station_id
        station_means = {}
        # from file AT/AT_5_48881_2019_timeseries.csv.gz and AT/AT_5_48900_2019_timeseries.csv
        station_id['concpm10'] = ['AT10002', 'AT52000']
        station_means['concpm10'] = [17.128, 15.113]
    
    
        # station_id['concso2'] = ['AT10002', 'AT52000']
        # station_id['conco3'] = ['XK0002A']
        # station_id['vmro3'] = ['XK0002A']
        # station_id['concno2'] = ['XK0002A']
        # station_id['concno2'] = ['AT31703']
        # station_id['concco'] = ['XK0002A']
        # station_id['concco'] = ['AT4S416']
        # station_id['concno'] = ['XK0002A']
        # station_id['concno'] = ['AT4S416']
        # station_id['concpm25'] = ['XK0002A']
    
        var_names_to_test = station_id.keys()
        for var_name in var_names_to_test:
            # r = reader()
            data = None
>           data = reader.read(vars_to_retrieve=[var_name])

tests/io/test_read_eea_aqerep_v2.py:69: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pyaerocom/io/read_eea_aqerep_base.py:579: in read
    station_data = self.read_file(_file, var_name=var_name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ReadEEAAQEREP_V2, filename = '/lustre/storeA/project/aerocom/aerocom1/AEROCOM_OBSDATA/EEA_AQeRep.v2/renamed/Revision.txt', var_name = 'concpm10'
vars_as_series = False

    def read_file(self, filename, var_name,
                  vars_as_series=False):
        """Read a single EEA file
    
        Note that there's only a single variable in the file
    
        Parameters
        ----------
        filename : str
            Absolute path to filename to read.
        var_name : str
            Name of variable in file.
        vars_as_series : bool
            If True, the data columns of all variables in the result dictionary
            are converted into pandas Series objects.
    
        Returns
        -------
        StationData
            Dict-like object containing the results.
    
        """
        if not var_name in self.PROVIDES_VARIABLES:
            raise ValueError('Invalid input variable {}'.format(var_name))
    
        # there's only one variable in the file
        aerocom_var_name = var_name
    
        # Iterate over the lines of the file
        self.logger.info("Reading file {}".format(filename))
        file_delimiter = self.FILE_COL_DELIM
        # this lists the data to keep from the original read string
        # this becomes a time series
        file_indexes_to_keep = [11, 13, 14, 15, 16]
        # used for line length control...
        max_file_index_to_keep = max(file_indexes_to_keep)
        # this is some header information
        header_indexes_to_keep = [0, 3, 8, 9, 10, 12, ]
        # These are the indexes with a time and are stored as np.datetime64
        time_indexes = [13, 14]
    
        # read the file
        # file_data = []
        # enable alternative reading of .gz files here to save space on the file system
        suffix = pathlib.Path(filename).suffix
        if suffix == '.gz':
            f_out = tempfile.NamedTemporaryFile(delete=False)
            with gzip.open(filename, 'r') as f_in:
                shutil.copyfileobj(f_in, f_out)
            filename = f_out.name
            f_in.close()
        with open(filename, 'r') as f:
            # read header...
            # Countrycode,Namespace,AirQualityNetwork,AirQualityStation,AirQualityStationEoICode,SamplingPoint,SamplingProcess,Sample,AirPollutant,AirPollutantCode,AveragingTime,Concentration,UnitOfMeasurement,DatetimeBegin,DatetimeEnd,Validity,Verification
            header = f.readline().lower().rstrip().split(file_delimiter)
            # create output dict
            data_dict = {}
            for idx in header_indexes_to_keep:
>               data_dict[header[idx]] = ''
E               IndexError: list index out of range

pyaerocom/io/read_eea_aqerep_base.py:282: IndexError
-------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------
  0%|          | 0/2 [00:00<?, ?it/s]
====================================================================== warnings summary ======================================================================
../../../anaconda.2018/envs/pyadev/lib/python3.9/site-packages/iris/fileformats/_ff.py:819
  /home/jang/anaconda.2018/envs/pyadev/lib/python3.9/site-packages/iris/fileformats/_ff.py:819: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    def _parse_binary_stream(file_like, dtype=np.float, count=-1):

../../../anaconda.2018/envs/pyadev/lib/python3.9/site-packages/pyke/knowledge_engine.py:28
  /home/jang/anaconda.2018/envs/pyadev/lib/python3.9/site-packages/pyke/knowledge_engine.py:28: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/stable/warnings.html
================================================================== short test summary info ===================================================================
FAILED tests/io/test_read_eea_aqerep_v2.py::test_read - IndexError: list index out of range
========================================================== 1 failed, 1 passed, 2 warnings in 0.52s ===========================================================
(pyadev) jang@pc5378:~/.../Python3/pyaerocom$ pytest tests/io/test_read_eea_aqerep_v2.py
==================================================================== test session starts =====================================================================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: /home/jang/data/Python3/pyaerocom, configfile: pytest.ini
plugins: anyio-3.3.0
collected 2 items                                                                                                                                            
run-last-failure: rerun previous 1 failure first

tests/io/test_read_eea_aqerep_v2.py ..                                                                                                                 [100%]

====================================================================== warnings summary ======================================================================
../../../anaconda.2018/envs/pyadev/lib/python3.9/site-packages/iris/fileformats/_ff.py:819
  /home/jang/anaconda.2018/envs/pyadev/lib/python3.9/site-packages/iris/fileformats/_ff.py:819: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    def _parse_binary_stream(file_like, dtype=np.float, count=-1):

../../../anaconda.2018/envs/pyadev/lib/python3.9/site-packages/pyke/knowledge_engine.py:28
  /home/jang/anaconda.2018/envs/pyadev/lib/python3.9/site-packages/pyke/knowledge_engine.py:28: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

tests/io/test_read_eea_aqerep_v2.py: 2091942 warnings
  /home/jang/data/Python3/pyaerocom/pyaerocom/io/read_eea_aqerep_base.py:313: DeprecationWarning: parsing timezone aware datetimes is deprecated; this will raise an error in the future
    data_dict[header[idx]][lineidx] = np.datetime64(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================ 2 passed, 2091944 warnings in 41.15s ============================================================```

Any idea what the problem might be? Is there a way to debug pytest runs?

**Comment by jgliss** [comment]
hmm, seems you need to register the observation ID, here is a snippet of what CI console outputs:

```
tests/io/test_read_eea_aqerep_v2.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pyaerocom/io/read_eea_aqerep_base.py:475: in get_file_list
    fp = os.path.join(self.data_dir, pattern)
pyaerocom/io/readungriddedbase.py:263: in data_dir
    self._data_dir = get_obsnetwork_dir(self.data_id)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obs_id = 'EEAAQeRep.v2'

    def get_obsnetwork_dir(obs_id):
        """Returns data path for obsnetwork ID
    
        Parameters
        ----------
        obs_id : str
            ID  of obsnetwork (e.g. AeronetSunV2Lev2.daily)
    
        Returns
        -------
        str
            corresponding directory from ``pyaerocom.const``
    
        Raises
        ------
        ValueError
            if obs_id is invalid
        IOError
            if directory does not exist
        """
        if not obs_id in const.OBSLOCS_UNGRIDDED:
>           raise ValueError(f"Observation network ID {obs_id} does not exist")
E           ValueError: Observation network ID EEAAQeRep.v2 does not exist

pyaerocom/io/helpers.py:235: ValueError
__________________________________ test_read ___________________________________

reader = ReadEEAAQEREP_V2

    @testdata_unavail
    def test_read(reader):
        from pyaerocom.ungriddeddata import UngriddedData
        from pyaerocom.stationdata import StationData
    
        # special station codes to test
        # not sure if these are really needed
    
        # station ids to test
        station_id = {}
        # respective mean for a station; index has to be the same as station_id
        station_means = {}
        # from file AT/AT_5_48881_2019_timeseries.csv.gz and AT/AT_5_48900_2019_timeseries.csv
        station_id['concpm10'] = ['AT10002', 'AT52000']
        station_means['concpm10'] = [17.128, 15.113]
    
    
        # station_id['concso2'] = ['AT10002', 'AT52000']
        # station_id['conco3'] = ['XK0002A']
        # station_id['vmro3'] = ['XK0002A']
        # station_id['concno2'] = ['XK0002A']
        # station_id['concno2'] = ['AT31703']
        # station_id['concco'] = ['XK0002A']
        # station_id['concco'] = ['AT4S416']
        # station_id['concno'] = ['XK0002A']
        # station_id['concno'] = ['AT4S416']
        # station_id['concpm25'] = ['XK0002A']
    
        var_names_to_test = station_id.keys()
        for var_name in var_names_to_test:
            # r = reader()
            data = None
>           data = reader.read(vars_to_retrieve=[var_name])

tests/io/test_read_eea_aqerep_v2.py:69: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pyaerocom/io/read_eea_aqerep_base.py:547: in read
    files = self.get_file_list(self.FILE_MASKS[var_name])
pyaerocom/io/read_eea_aqerep_base.py:475: in get_file_list
    fp = os.path.join(self.data_dir, pattern)
pyaerocom/io/readungriddedbase.py:263: in data_dir
    self._data_dir = get_obsnetwork_dir(self.data_id)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
```

**Comment by jgliss** [comment]
@jgriesfeller I think you need to register the new obsdataset in the header of class `TestDataAccess` (file pyaerocom/testdata_access.py), see class attrs `ADD_PATHS` and `_UNGRIDDED_READERS`.

**Comment by jgriesfeller** [comment]
Some of the test seem to work now.
Unfortunately I mixed up some development for the reading in here. In wanted to create a seperate PR. 
But since @hansbrenna needs the corrections in the EEA reading, it would be good to review that together.

**Comment by jgriesfeller** [comment]
addresses also #405

File: pyaerocom/io/read_eea_aqerep_base.py [hunk 1]
```diff
@@ -272,11 +272,27 @@ def read_file(self, filename, var_name,
...
+                    os.remove(f_out.name)
+                    raise EEAv2FileError(
+                        f'Found corrupt file {filename}. consider deleteing it')
+
             # create output dict
+            if len(header) < max_file_index_to_keep:
+                if suffix == '.gz':
+                    os.remove(f_out.name)
+                raise EEAv2FileError(
+                    f'Found corrupt file {filename}. consider deleteing it')
```
Comment (jgliss) [inline pyaerocom/io/read_eea_aqerep_base.py:294]:
```suggestion
                    f'Found corrupt file {filename}. consider deleting it')
```

File: pyaerocom/io/read_eea_aqerep_base.py [hunk 2]
```diff
@@ -272,11 +272,27 @@ def read_file(self, filename, var_name,
                 shutil.copyfileobj(f_in, f_out)
             filename = f_out.name
             f_in.close()
```
Comment (jgliss) [inline pyaerocom/io/read_eea_aqerep_base.py:274]:
```suggestion

```

I think f_in is closed automatically via the contextmanager.

**Review by jgliss — APPROVED** [review]
Looks good as far as I can see, you might want to check the few minor comments and suggestions I made.

**Comment by jgliss** [comment]
@jgriesfeller just re-triggered CI

File: tests/io/test_read_eea_aqerep_v2.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            try:
+                stat_data = data[statid]
+                # It makes no sense to test this for every station
+                if stat_idx == 1:
+                    assert isinstance(stat_data, StationData)
+
+                assert stat_data[var_name].mean() == station_means[var_name][stat_idx]
+            except:
+                print('failed test var {}'.format(var_name))
+                pass
```
Comment (avaldebe) [inline tests/io/test_read_eea_aqerep_v2.py:83]: @jgriesfeller 
What is this?
as far as I can tell this the test is failing.
Can you have a look on the current main-dev?

**Pull request merged**